libcaf_test/caf/test/reporter.cpp

```diff
@@ -373,8 +373,8 @@ class default_reporter : public reporter {
               ' ', indent_, location.file_name(), location.line(), msg);
   }
 
-  void info(std::string_view msg,
-            const detail::source_location& location) override {
+  void print_info(std::string_view msg,
```

Comment:
I had initially thought of making a single `print` function that can call `print_info`, `print_debug` or `print_error` based on the verbosity that is passed in the function. But, I think that this does not provide any benefit to the users of the library or in the code base as we still have to use the copy and pasted `print_*` libraries privately. 

We can probably keep a single `print` function that prints based on verbosity by dynamically configuring the print string in a single function. But, there is a small drawback to that approach, which is that it will be much harder to add different type of print functionalities (like trace) in the future. Let me know your preference regarding this.

Reply:
Yeah, good point. Trace wouldn't play nice with a single `print` function since we need enter and leave events.

Then let's at least reduce the code duplication with a private print function, e.g.:

```c++
  void print_debug(std::string_view msg,
                   const detail::source_location& location) override {
    print_impl(CAF_LOG_LEVEL_DEBUG, 'C', "debug", msg, location);
  }
```

Whereas `B` is the color code (cyan in this case), and `debug` is the text for the message. I hope the rest is self-explanatory regarding how to implement `print_impl`.